source/Test.cpp

```diff
@@ -87,21 +87,111 @@ Test::TestStep::TestStep(Type stepType) : stepType(stepType)
 
 void Test::LoadSequence(const DataNode &node)
 {
+	if(!steps.empty())
+	{
+		status = Status::BROKEN;
+		node.PrintTrace("Error: duplicate sequence keyword");
+		return;
+	}
+	
 	for(const DataNode &child: node)
 	{
 		const string &typeName = child.Token(0);
 		auto it = find_if(STEPTYPE_TO_TEXT.begin(), STEPTYPE_TO_TEXT.end(),
 			[&typeName](const std::pair<TestStep::Type, const string> &e) {
 				return e.second == typeName;
 			});
-		if(it != STEPTYPE_TO_TEXT.end())
-			steps.emplace_back(it->first);
-		else
+		if(it == STEPTYPE_TO_TEXT.end())
 		{
 			status = Status::BROKEN;
 			child.PrintTrace("Unsupported step type (" + ExpectedOptions(STEPTYPE_TO_TEXT) + "):");
 			// Don't bother loading more steps once broken.
-			break;
+			return;
+		}
+		
+		steps.emplace_back(it->first);
+		TestStep &step = steps.back();
+		switch(step.stepType)
+		{
+			case TestStep::Type::APPLY:
+			case TestStep::Type::ASSERT:
+				step.conditions.Load(child);
+				break;
+			case TestStep::Type::BRANCH:
+				if(child.Size() < 2)
+				{
+					status = Status::BROKEN;
+					child.PrintTrace("Error: branch without jump location");
+					return;
+				}
+				step.jumpOnTrueTarget = child.Token(1);
+				if(child.Size() > 2)
+					step.jumpOnFalseTarget = child.Token(2);
+				step.conditions.Load(child);
+				break;
+			case TestStep::Type::INJECT:
+				if(child.Size() < 2)
+				{
+					status = Status::BROKEN;
+					child.PrintTrace("Error: inject without data identifier");
+					return;
+				}
+				else
+					step.nameOrLabel = child.Token(1);
+			case TestStep::Type::INPUT:
+				child.PrintTrace("Error: Not yet implemented step type input");
+				status = Status::BROKEN;
+				return;
+			case TestStep::Type::INVALID:
+				status = Status::BROKEN;
+				return;
+			case TestStep::Type::LABEL:
+				if(child.Size() < 2)
+					child.PrintTrace("Ignoring empty label");
+				else
+				{
+					step.nameOrLabel = child.Token(1);
+					if(jumpTable.find(step.nameOrLabel) != jumpTable.end())
+					{
+						child.PrintTrace("Error: duplicate label");
+						status = Status::BROKEN;
+						return;
+					}
+					else
+						jumpTable[step.nameOrLabel] = steps.size()-1;
+				}
+				break;
+			case TestStep::Type::NAVIGATE:
+				child.PrintTrace("Error: Not yet implemented step type navigation");
+				status = Status::BROKEN;
+				return;
+			case TestStep::Type::WATCHDOG:
+				if(child.Size() < 2)
+					step.watchdog = 0;
+				else
+					step.watchdog = child.Value(1);
```

Comment:
For most of the game code, we do DataNode size checks in the "meets criteria" phrasing first.
```suggestion
				step.watchdog = child.Size() >= 2 ? child.Token(1) : 0;
```